File: DesktopSample/Advanced/PageAndRDL/CustomDataProvider/VB.NET/CustomDataProvider/CSVDataProvider/CsvDataProviderFactory.vb
Imports System.Data.Common

''' <summary>
''' .NETフレームワークのCSVデータプロバイダの<see cref="DataProviderFactory"/>を実装する方法を提供します。
''' </summary>
Public Class CsvDataProviderFactory
	Inherits DbProviderFactory

	''' <summary>
	''' <see cref="CsvCommand"/>の新しいインスタンスを返します。
	''' </summary>
	Public Overrides Function CreateCommand() As DbCommand
		Return New CsvCommand()
	End Function

	''' <summary>
	''' <see cref="CsvConnection"/>の新しいインスタンスを返します。
	''' </summary>
	Public Overrides Function CreateConnection() As DbConnection
		Return New CsvConnection()
	End Function
End Class

File: DesktopSample/Advanced/PageAndRDL/CustomDataProvider/C#/CustomDataProvider/CSVDataProvider/CsvDataProviderFactory.cs
using ActiveReports.Samples.CustomDataProvider.CsvDataProvider;
using System.Data.Common;

namespace ActiveReports.Samples.CustomDataProvider.CSVDataProvider
{
	/// <summary>
	/// .NETフレームワークのCSVデータプロバイダの<see cref="DataProviderFactory"/>を実装する方法を提供します。
	/// </summary>
	public class CsvDataProviderFactory : DbProviderFactory
	{
		/// <summary>
		/// <see cref="CsvCommand"/>の新しいインスタンスを返します。
		/// </summary>
		public override DbCommand CreateCommand()
		{
			return new CsvCommand();
		}

		/// <summary>
		/// <see cref="CsvConnection"/>の新しいインスタンスを返します。
		/// </summary>
		public override DbConnection CreateConnection()
		{
			return new CsvConnection();
		}
	}
}
